diff --git a/include/jwt/algorithm.hpp b/include/jwt/algorithm.hpp
index 0e3b843..1156e6a 100644
--- a/include/jwt/algorithm.hpp
+++ b/include/jwt/algorithm.hpp
@@ -64,6 +64,8 @@ using verify_func_t = verify_result_t (*) (const jwt::string_view key,
                                            const jwt::string_view head,
                                            const jwt::string_view jwt_sign);
 
+verify_result_t is_secret_a_public_key(const jwt::string_view secret);
+
 namespace algo {
 
 //Me: TODO: All these can be done using code generaion.
