diff --git a/quazip/minizip_crypt.h b/quazip/minizip_crypt.h
index 2e833f7..ea9d277 100644
--- a/quazip/minizip_crypt.h
+++ b/quazip/minizip_crypt.h
@@ -90,13 +90,14 @@ static void init_keys(const char* passwd,unsigned long* pkeys,const z_crc_t FAR
 #    define ZCR_SEED2 3141592654UL     /* use PI as default pattern */
 #  endif
 
-static int crypthead(passwd, buf, bufSize, pkeys, pcrc_32_tab, crcForCrypting)
-    const char *passwd;         /* password string */
-    unsigned char *buf;         /* where to write header */
-    int bufSize;
-    unsigned long* pkeys;
-    const z_crc_t FAR * pcrc_32_tab;
-    unsigned long crcForCrypting;
+static int crypthead(
+    const char *passwd,         /* password string */
+    unsigned char *buf,         /* where to write header */
+    int bufSize,
+    unsigned long* pkeys,
+    const z_crc_t FAR * pcrc_32_tab,
+    unsigned long crcForCrypting
+)
 {
     int n;                       /* index in random header */
     int t;                       /* temporary */
